addon/components/offering-form.js

```diff
@@ -98,37 +93,29 @@ export default Component.extend(ValidationErrorDisplay, Validations, {
   loaded: false,
   'data-test-offering-form': true,
 
-  associatedSchools: computed('cohorts.[]', function(){
-    return new RSVPPromise(resolve => {
-      const cohorts = this.get('cohorts');
-      if (isEmpty(cohorts)) {
-        resolve([]);
-      } else {
-        map(cohorts.toArray(), cohort => {
-          return cohort.get('school');
-        }).then(cohortSchools => {
-          const schools = [];
-          schools.pushObjects(cohortSchools);
-          resolve(schools.uniq().toArray());
-        });
-      }
-    });
-  }),
-  availableInstructorGroups: computed('associatedSchools.[]', function(){
-    return new RSVPPromise(resolve => {
-      this.get('associatedSchools').then(associatedSchools => {
-        map(associatedSchools, school => {
-          return school.get('instructorGroups');
-        }).then(allInstructorGropus => {
-          const flat = allInstructorGropus.reduce((flattened, obj) => {
-            return flattened.pushObjects(obj.toArray());
-          }, []);
-
-          resolve(flat);
-        });
+  associatedSchools: computed('cohorts.[]', async function(){
+    const cohorts = this.get('cohorts');
+    if (isEmpty(cohorts)) {
+      return [];
+    } else {
+      const cohortSchools = map(cohorts.toArray(), cohort => {
+        return cohort.get('school');
       });
+      const schools = [];
+      schools.pushObjects(cohortSchools);
+      return schools.uniq().toArray();
+    }
+  }),
+  availableInstructorGroups: computed('associatedSchools.[]', async function(){
+    const associatedSchools = await this.get('associatedSchools');
+    const allInstructorGropus = await map(associatedSchools, school => {
+      return school.get('instructorGroups');
     });
+    return allInstructorGropus.reduce((flattened, obj) => {
```

Comment:
While you're at it this can now almost certainly be replace with the new [Array.flat()](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/flat).

Reply:
the inner arrays are not real arrays, they're DS.PromiseArray objects. leaving the current code as-is, since it dumps these out into proper arrays in the process of flattening.

Reply:
👍